### PR TITLE
Use necessary WebGL numeric precision

### LINKED ISSUE #15280: Harden WebGL numeric precision
WebGL glyph shaders request medium floating-point precision even where coordinate transforms require the wider range and accuracy of high precision. The WebGL greatest-common-divisor helper also does not safely handle empty, zero-valued, or signed inputs used while preparing dash data.

## Impact

Large or finely spaced coordinates can produce avoidable rendering artifacts on affected GPUs, while degenerate dash inputs can yield invalid numeric results and destabilize WebGL dash processing.

## Desired behavior

WebGL vertex shaders should use high precision, fragment shaders should prefer high precision with a portable medium-precision fallback, and the generated shader pairs should compile and link in a real WebGL context. The GCD helper should return stable non-negative results for empty, zero-valued, and signed inputs.

## Minimal reproducer

A zero-length dash pattern should normalize to a solid line. The old WebGL GCD path can instead produce invalid arithmetic while constructing the dash texture.

```python
from bokeh.plotting import figure, show

p = figure(width=700, height=300, output_backend="webgl")
p.line([0, 1, 2, 3], [0, 1, 0, 1], line_width=5, line_dash=[0, 0])
show(p)
```

Shader precision artifacts are hardware-dependent and are most likely on GPUs whose fragment `highp` support is limited.

### PR DESCRIPTION
**Pull request by bryevdv**
[codex assisted]

- [x] issues: fixes #15280
- [x] tests added / passed

## Summary

- request high floating-point precision in WebGL vertex shaders
- prefer high precision in fragment shaders while retaining a portable `mediump` fallback
- make the WebGL GCD helper robust for empty, zero-valued, and signed inputs
- add focused unit coverage for shader precision declarations, real WebGL shader compilation/linking, and GCD edge cases

### CODE REVIEW

File: bokehjs/test/unit/models/glyphs/webgl/math.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {expect} from "#framework/assertions"
+
+import {gcd} from "@bokehjs/models/glyphs/webgl/utils/math"
+
+describe("WebGL math utilities", () => {
+  it("should return zero for empty and all-zero inputs", () => {
+    expect(gcd([])).to.be.equal(0)
```
Comment (muendlein) [inline bokehjs/test/unit/models/glyphs/webgl/math.ts:7]: Should this really return 0 or rather null/undefined?

File: bokehjs/test/unit/models/glyphs/webgl/math.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    expect(gcd([0])).to.be.equal(0)
```
Comment (muendlein) [inline bokehjs/test/unit/models/glyphs/webgl/math.ts:8]: Probably good to also have tests with single items that are non zero (both positive and negative)

File: bokehjs/test/unit/models/glyphs/webgl/math.ts (hunk 1 above)
Comment (bryevdv) [reply bokehjs/test/unit/models/glyphs/webgl/math.ts:7]: There's various arguments for this, but I will just appeal to authority here:
```python
In [1]: import math

In [2]: math.gcd()
Out[2]: 0
```

File: bokehjs/test/unit/models/glyphs/webgl/math.ts (hunk 2 above)
Comment (bryevdv) [reply bokehjs/test/unit/models/glyphs/webgl/math.ts:8]: 5b2f70de89fe5550d0909fe386baea708dce50dc

**Review by muendlein — APPROVED** [review]
